### PR TITLE
Initial JSON columns sample

### PR DESCRIPTION
**Pull request by ajcvickers**

### CODE REVIEW

File: entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md [hunk 1]
```diff
@@ -19,6 +19,414 @@ EF7 is also available as [daily builds](https://github.com/dotnet/efcore/blob/ma
...
+-->
+[!code-csharp[PostMetadataAggregate](../../../../samples/core/Miscellaneous/NewInEFCore7/BlogsContext.cs?name=PostMetadataAggregate)]
+
+This aggregate type contains several nested types and collections. Calls to `OwnsOne` and `OwnsMany` are used to map this aggregate type:
+
+<!--
+        modelBuilder.Entity<Post>().OwnsOne(
+            post => post.Metadata, ownedNavigationBuilder =>
+            {
+                ownedNavigationBuilder.ToJson();
```
Comment (maumar) [inline entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md:256]: might be worth mentioning that you only need to call ToJson on the top level. It's been said/implied in the ContactDetails example above, but that one didn't have any children, here it's more visible.

File: samples/core/Miscellaneous/NewInEFCore7/JsonColumnsSample.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        Console.WriteLine();
+        foreach (var address in orderedAddresses)
+        {
+            Console.WriteLine(address);
+        }
+
+        Console.WriteLine();
+
+        // Since query below cannot use Include
+        // Issue: https://github.com/dotnet/efcore/issues/28808
```
Comment (maumar) [inline samples/core/Miscellaneous/NewInEFCore7/JsonColumnsSample.cs:82]: this is now fixed

File: entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md [hunk 3]
```diff
@@ -19,6 +19,414 @@ EF7 is also available as [daily builds](https://github.com/dotnet/efcore/blob/ma
 
 EF7 targets .NET 6, and so can be used with either [.NET 6 (LTS)](https://dotnet.microsoft.com/download/dotnet/6.0) or [.NET 7](https://dotnet.microsoft.com/download/dotnet/7.0).
 
+## JSON Columns
+
+Most relational databases support columns that contain JSON documents. The JSON in these columns can be drilled into with queries. This allows, for example, filtering and sorting by the elements of the documents, as well as projection of elements out of the documents into results. JSON columns allow relational databases to take on some of the characteristics of document databases, creating a useful hybrid between the two.
+
+EF7 contains provider-agnostic support for JSON columns, with an implementation for SQL Server. This support allows mapping of aggregates built from .NET types to JSON documents. Normal LINQ queries can be used on the aggregates, and these will be translated to the appropriate query constructs needed to drill into the JSON. EF7 also supports updating and saving changes to the JSON documents.
```
Comment (maumar) [inline entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md:26]: worth adding an example for update?

File: entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md (hunk 1 above)
Comment (ajcvickers) [reply entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md:256]: Will do in the next PR.

File: samples/core/Miscellaneous/NewInEFCore7/JsonColumnsSample.cs (hunk 2 above)
Comment (ajcvickers) [reply samples/core/Miscellaneous/NewInEFCore7/JsonColumnsSample.cs:82]: Will update in the next PR.

File: entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md (hunk 3 above)
Comment (ajcvickers) [reply entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md:26]: Was waiting for partial updates before adding an example here.

**Comment by ajcvickers** [comment]
@dotnet/efteam Can someone approve this so I can merge and rebase more recent work on it? Will make requested changes in the next update to this area.

File: entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md [hunk 4]
```diff
@@ -19,6 +19,414 @@ EF7 is also available as [daily builds](https://github.com/dotnet/efcore/blob/ma
...
+
+> [!NOTE]
+> SQLite support for JSON is [planned for post EF7](https://github.com/dotnet/efcore/issues/28816). The PostgreSQL and Pomelo MySQL providers already contain some support for JSON columns. We will be working with the authors of those providers to align JSON support across all providers.
```
Comment (roji) [inline entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md:29]: I dunno, I heard the PG guy is annoying

File: entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md [hunk 5]
```diff
@@ -19,6 +19,414 @@ EF7 is also available as [daily builds](https://github.com/dotnet/efcore/blob/ma
...
+            .ToListAsync();
+-->
+[!code-csharp[AuthorsInChigley](../../../../samples/core/Miscellaneous/NewInEFCore7/JsonColumnsSample.cs?name=AuthorsInChigley)]
+
+This query generates the following SQL when using SQL Server:
+
+```sql
+SELECT [a].[Id], [a].[Name], JSON_QUERY([a].[Contact],'$')
+FROM [Authors] AS [a]
+WHERE CAST(JSON_VALUE([a].[Contact],'$.Address.City') AS nvarchar(max)) = N'Chigley'
```
Comment (roji) [inline entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md:352]: Hmm... Are we sure that we need to systematically add the CAST to nvarchar(max) everywhere where we do JSON_VALUE? At least in this case (and below) that doesn't seem necessary. In other words, 

/cc @maumar

File: entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md [hunk 6]
```diff
@@ -19,6 +19,414 @@ EF7 is also available as [daily builds](https://github.com/dotnet/efcore/blob/ma
...
+
+```sql
+SELECT [a].[Name], CAST(JSON_VALUE([p].[Metadata],'$.Views') AS int), JSON_QUERY([p].[Metadata],'$.TopSearches'), [p].[Id], JSON_QUERY([p].[Metadata],'$.Updates')
+FROM [Posts] AS [p]
+LEFT JOIN [Authors] AS [a] ON [p].[AuthorId] = [a].[Id]
+WHERE CAST(JSON_VALUE([p].[Metadata],'$.Views') AS int) > 3000
+```
+
+> [!NOTE]
+> More complex queries involving JSON collections require `jsonpath` support. Vote for [Support jsonpath querying](https://github.com/dotnet/efcore/issues/28616) if this is something you are interested in.
```
Comment (roji) [inline entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md:428]: Maybe add a note to the [SQL Server JSON index docs](https://docs.microsoft.com/en-us/sql/relational-databases/json/index-json-data?view=sql-server-ver16)?

**Review by roji — APPROVED** [review]

**Pull request merged**